aarch64/src/trap.S

```diff
@@ -0,0 +1,180 @@
+.section .text
+
+.equ	SYNC_INVALID_EL1t,	0
+.equ	IRQ_INVALID_EL1t,	1
+.equ	FIQ_INVALID_EL1t,	2
+.equ	ERROR_INVALID_EL1t,	3
+
+.equ	SYNC_INVALID_EL1h,	4
+.equ	IRQ_INVALID_EL1h,	5
+.equ	FIQ_INVALID_EL1h,	6
+.equ	ERROR_INVALID_EL1h,	7
+
+.equ	SYNC_INVALID_EL0_64,	8
+.equ	IRQ_INVALID_EL0_64,	9
+.equ	FIQ_INVALID_EL0_64,	10
+.equ	ERROR_INVALID_EL0_64,	11
+
+.equ	SYNC_INVALID_EL0_32,	12
+.equ	IRQ_INVALID_EL0_32,	13
+.equ	FIQ_INVALID_EL0_32,	14
+.equ	ERROR_INVALID_EL0_32,	15
+
+.macro	ventry	label
+.align	7
```

Comment:
Will this align on 128-byte boundaries?

Reply:
Yes, though changed to balign for a bit more clarity.